internal/k8s/container.go

```diff
@@ -40,32 +41,32 @@ func WaitForContainerReady(ctx context.Context, client Client, pod *v1.Pod, ref
 				continue
 			}
 
-			ready, err := waitForContainerReadyHelper(pod, ref)
+			cID, err := waitForContainerReadyHelper(pod, ref)
 			if err != nil {
-				return err
-			} else if ready {
-				return nil
+				return "", err
+			} else if cID != "" {
+				return cID, nil
 			}
 		}
 	}
-	return nil
+	return "", nil
```

Comment:
hmm it feels weird me to return a nil error here? even tho it's a case we'll theoretically never hit...? 🤷‍♀️

Reply:
what do you think it should return?